x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx

```diff
@@ -118,19 +110,46 @@ export function TimeseriesChart({
       ? anomalyChartTimeseries?.boundaries
       : []),
     ...(anomalyChartTimeseries?.scores ?? []),
-  ];
+  ]
+    // Sorting series so that area type series are before line series
+    // This is a workaround so that the legendSort works correctly
+    .sort(
+      isComparingExpectedBounds
+        ? (prev, curr) => prev.type.localeCompare(curr.type)
+        : undefined
+    );
+
+  const xValues = timeseries.flatMap(({ data }) => data.map(({ x }) => x));
+
+  const timeZone = getTimeZone(core.uiSettings);
+
+  const min = Math.min(...xValues);
+  const max = Math.max(
+    ...xValues,
+    // If the last data point of timeseries ends before (picked end time + bucket_span)
+    // we need to extend the x max domain to the last available data point for the expected bounds
+    // for the area chart to show up correctly
+    // See https://github.com/elastic/elastic-charts/issues/1685
+    isComparingExpectedBounds
+      ? anomalyChartTimeseries?.boundaries[0]?.data?.slice(-1)[0]?.x ?? 0
+      : -Infinity
+  );
+  const xFormatter = niceTimeFormatter([min, max]);
+
   const xDomain = isEmpty ? { min: 0, max: 1 } : { min, max };
 
-  const legendSort = (a: SeriesIdentifier, b: SeriesIdentifier) => {
-    // Using custom legendSort here when comparing expected bounds
-    // because by default elastic-charts will show legends for expected bounds first
-    // but for consistency, we are making `Expected bounds` last
-    if ((a as XYChartSeriesIdentifier)?.specId === expectedBoundsTitle)
-      return -1;
-    if ((b as XYChartSeriesIdentifier)?.specId === expectedBoundsTitle)
-      return -1;
-    return 1;
-  };
+  // Using custom legendSort here when comparing expected bounds
+  // because by default elastic-charts will show legends for expected bounds first
+  // but for consistency, we are making `Expected bounds` last
+  const legendSort = isComparingExpectedBounds
+    ? (a: SeriesIdentifier, b: SeriesIdentifier) => {
+        if ((a as XYChartSeriesIdentifier)?.specId === expectedBoundsTitle)
+          return -1;
+        if ((b as XYChartSeriesIdentifier)?.specId === expectedBoundsTitle)
+          return -1;
+        return 1;
+      }
+    : undefined;
```

Comment:
Afaict this patches a relatively minor bug that will hopefully be fixed soon by the charts team. 
In the name of reducing complexity I think we can remove this. Did someone specifically ask for this to be fixed?

Reply:
I think for consistency and user experience this is better, albeit it is a workaround. If not, it will be very confusing when the legends keep switching when changing between time comparisons and expected bounds (e.g. 'Average' would be some time first and the other times last). Personally I think it's okay to keep this for now because we don't necessarily have a timeline for when it will be fixed on the charts side.